### ISSUE TITLE
Issue with configuring javafxports

### ISSUE THREAD

**Issue by Xerus2000** [gradle/kotlin-dsl#788]
```
buildscript {
	repositories {
		jcenter()
	}
	dependencies {
		classpath("org.javafxports:jfxmobile-plugin:1.3.8")
	}
}

jfxmobile {
	downConfig = DownConfiguration(project).apply {
		plugins("storage", "settings")
	}
	android {
		manifest = "src/android/AndroidManifest.xml"
	}
}
```

### Expected Behavior
This buildscript compiles without issues

### Current Behavior
Upon compiling the buildscript, I get an `Unresolved reference: android`

### Context
When moving from groovy to kotlin-dsl, I tried porting the jfxmobile configuration, got the downConfig working (as seen above), but not the android block. This is the original in groovy:

```
buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'org.javafxports:jfxmobile-plugin:1.3.8'
    }
}
apply plugin: 'org.javafxports.jfxmobile'

jfxmobile {
    android {
        manifest = 'src/android/AndroidManifest.xml'

        compileSdkVersion = 26
        minSdkVersion = 26
    }
    downConfig {
        plugins 'storage', 'settings'
    }
}
```

**Comment by eskatos** [comment]
Hi @Xerus2000,
Thank you for the report.

In order to get static accessors to dynamic Gradle model elements you need to apply your plugins using the `plugins {}` block or from a parent project. See [this document](https://github.com/gradle/kotlin-dsl/blob/master/doc/getting-started/Configuring-Plugins.md) for more information.

The `org.javafxports.jfxmobile` plugin is not published to the plugin portal nor has plugin markers published ([javafxports/javafxmobile-plugin#80](https://bitbucket.org/javafxports/javafxmobile-plugin/issues/80/gradle-plugin-portal-and-new-plugins)). As described in the document linked above this can be worked around.

Moreover, the `jfxmobile` Project extension created by this plugin itself has dynamic extensions, which are not yet configured easily, see https://github.com/gradle/kotlin-dsl/issues/457. Again this can be worked around.

Here is a working build:

```kotlin
// settings.gradle.kts

// Work around https://bitbucket.org/javafxports/javafxmobile-plugin/issues/80/gradle-plugin-portal-and-new-plugins
pluginManagement {
    repositories {
        gradlePluginPortal()
        jcenter()
    }
    resolutionStrategy {
        eachPlugin {
            if(requested.id.id == "org.javafxports.jfxmobile") {
                useModule("org.javafxports:jfxmobile-plugin:${requested.version}")
            }
        }
    }
}
```

```kotlin
// build.gradle.kts

import org.javafxports.jfxmobile.plugin.DownConfiguration
import org.javafxports.jfxmobile.plugin.JFXMobileExtension
import org.javafxports.jfxmobile.plugin.android.AndroidExtension

plugins {
    id("org.javafxports.jfxmobile") version "1.3.10"
}

application {
    mainClassName = "some.Thing"
}

jfxmobile {
    downConfig = DownConfiguration(project).apply {
        plugins("storage", "settings")
    }
    android {
        manifest = "src/android/AndroidManifest.xml"
    }
}

// Work around https://github.com/gradle/kotlin-dsl/issues/457
fun JFXMobileExtension.android(action: AndroidExtension.() -> Unit) =
    (this as ExtensionAware).extensions.configure(AndroidExtension::class.java, action)
```

HTH

**Comment by Xerus2000** [comment]
Thanks for your fast assistance, worked perfectly! I was pretty sure there was a workaround but it's hard to search for stuff like that. 
Since there seem to be many issues around that and other common problems that are not yet implemented, wouldn't it be appropriate to have a "Common workarounds" section in the wiki?
